vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,539 @@
+See the "Default Settings" section of the [mathematical model](model_definition.html) vignette for more details.
+```{r}
+ref_time_allocation <- allocate_reference_times(rep_tri)
+n_history_delay <- ref_time_allocation$n_history_delay
+n_retrospective_nowcasts <- ref_time_allocation$n_retrospective_nowcasts
+```
+
+# Run the `baselinenowcast` workflow
+
+First, we generate the point estimate of the nowcast as a "filled in" reporting triangle.
+See the sections "Estimating the delay distribution from a reporting triangle" and "Point nowcast generation" of the [mathematical model](model_definition.html) for more details.
+```{r}
+pt_nowcast_matrix <- estimate_and_apply_delay(rep_tri, n = n_history_delay)
+```
+
+Next, we estimate and apply uncertainty by generating retrospective nowcasts and comparing against later observed data.
+See the section "Uncertainty quantification" of the [mathematical_model](model_definition.html) for more details on assumptions being made here and a further description of how the uncertainty is quantified using past nowcast errors.
+```{r}
+nowcast_draws_df <- estimate_and_apply_uncertainty(
+  pt_nowcast_matrix,
+  rep_tri,
+  n_history_delay = n_history_delay,
+  n_retrospective_nowcasts = n_retrospective_nowcasts
+)
+
+head(nowcast_draws_df)
+```
+
+# Summarise and plot the nowcast
+
+Now that we have a dataframe of probabilistic nowcast draws, which represent our predicted "final" case count at each reference time, we can summarise them to compute the 50th and 95th percent prediction intervals, which are used to communicate uncertainty in the nowcast.
+```{r}
+nowcast_summary_df <-
+  nowcast_draws_df |>
+  group_by(time) |>
+  summarise(
+    median = median(pred_count),
+    q50th_lb = quantile(pred_count, 0.25),
+    q50th_ub = quantile(pred_count, 0.75),
+    q95th_lb = quantile(pred_count, 0.025),
+    q95th_ub = quantile(pred_count, 0.975)
+  )
+```
+
+Next, we will join the initially reported cases and the held out, true final cases to our probabilistic nowcast.
+The final cases will be used to evaluate the performance of our nowcast given the right-truncated initial reports.
+```{r}
+training_df_by_ref_date <- training_df |>
+  filter(report_date <= nowcast_date) |>
+  group_by(reference_date) |>
+  summarise(initial_count = sum(count)) |>
+  mutate(time = row_number())
+
+eval_data <- long_df |>
+  filter(
+    delay <= max_delay,
+    reference_date <= nowcast_date
+  ) |>
+  group_by(reference_date) |>
+  summarise(final_count = sum(count)) |>
+  mutate(time = row_number()) |>
+  select(time, final_count)
+
+nowcast_w_data <- nowcast_summary_df |>
+  left_join(training_df_by_ref_date,
+    by = "time"
+  ) |>
+  left_join(eval_data,
+    by = "time"
+  )
+head(nowcast_w_data)
+```
+
+## Plot against later observed "final" data
+Next we will make a plot of the nowcasted "final" cases compared to the initially reported, right-truncated cases and the true "final" reports.
```

Comment:
Tell me about what I see is this good. Is this useful for public health why. What would happen if I didn't do it. Link back to the need we talked about at the beginning. 

I think we need to visualise the data earlier to highlight the problem as people may have died from boredom at this point. Suggest visualising after cleaning as a motivation for doing nowcasting as at the moment its tell not show which is bad

Reply:
Ok I added a plot of the initial data in red right after we create `training_df` which shows the rapidly declining recent trends

Reply:
Plus motivation etc for nowcasting